shutil.move stub does not account for bytes paths

Just encountered this.

Consider this valid script:
```python
import os
from pathlib import Path
import shutil
from typing import AnyStr


class DummyPathLike(os.PathLike[AnyStr]):
    """Since pathlib.Path can't accept bytes itself."""

    def __init__(self, path: AnyStr) -> None:
        self._path: AnyStr = path

    def __fspath__(self) -> AnyStr:
        """Return the file system path representation of the object."""
        return self._path


Path("test1.txt").touch()

# Both str
shutil.move("test1.txt", "test2.txt")

# Both bytes
shutil.move(b"test2.txt", b"test1.txt")

# Both Pathlike[str]
shutil.move(Path("test1.txt"), Path("test2.txt"))

# Both Pathlike[bytes]
# (N.B. pathlib.Path doesn't accept bytes so made dummy bytes pathlike for demo)
shutil.move(DummyPathLike(b"test2.txt"), DummyPathLike(b"test1.txt"))

# There's no constraints between the src and dst path types so no need for type vars.
shutil.move("test1.txt", DummyPathLike(b"test2.txt"))

Path("test2.txt").unlink()
```

mypy gives the following false positives:
```
type "bytes"; expected "Union[str, PathLike[str]]"
test.py:24: error: Argument 2 to "move" has incompatible type "bytes"; expected "Union[str, PathLike[str]]"
test.py:31: error: Argument 1 to "DummyPathLike" has incompatible type "bytes"; expected "str"
test.py:34: error: Argument 1 to "DummyPathLike" has incompatible type "bytes"; expected "str"
Found 4 errors in 1 file (checked 1 source file)
```

Should be a quick fix to switch StrPath with StrOrBytesPath in the stub so will pick this up now 👍

--- a/stdlib/shutil.pyi
+++ b/stdlib/shutil.pyi
@@ -49,11 +49,11 @@ def rmtree(path: bytes | StrPath, ignore_errors: bool = ..., onerror: Callable[[
 _CopyFn = Union[Callable[[str, str], None], Callable[[StrPath, StrPath], None]]
 
 if sys.version_info >= (3, 9):
-    def move(src: StrPath, dst: StrPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
+    def move(src: StrOrBytesPath, dst: StrOrBytesPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
 
 else:
     # See https://bugs.python.org/issue32689
-    def move(src: str, dst: StrPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
+    def move(src: str | bytes, dst: StrOrBytesPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
 
 class _ntuple_diskusage(NamedTuple):
     total: int

--- a/stdlib/shutil.pyi
+++ b/stdlib/shutil.pyi
@@ -48,12 +48,15 @@ def rmtree(path: bytes | StrPath, ignore_errors: bool = ..., onerror: Callable[[
 
 _CopyFn = Union[Callable[[str, str], None], Callable[[StrPath, StrPath], None]]
 
+# N.B. shutil.move appears to take bytes arguments, however,
+# this does not work when dst is (or is within) an existing directory.
+# (#6832)
 if sys.version_info >= (3, 9):
-    def move(src: StrOrBytesPath, dst: StrOrBytesPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
+    def move(src: StrPath, dst: StrPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
 
 else:
     # See https://bugs.python.org/issue32689
-    def move(src: str | bytes, dst: StrOrBytesPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
+    def move(src: str, dst: StrPath, copy_function: _CopyFn = ...) -> _PathReturn: ...
 
 class _ntuple_diskusage(NamedTuple):
     total: int

